app/Controllers/authController.php

```diff
@@ -155,10 +156,15 @@ public function formLoginAction() {
 				                  ' user=' . $username .
 				                  ', nonce=' . $nonce .
 				                  ', c=' . $challenge);
-				Minz_Request::bad(
-					_t('feedback.auth.login.invalid'),
-					array('c' => 'auth', 'a' => 'login')
-				);
+
+				header('HTTP/1.1 403 Forbidden');
+				Minz_Session::_param('POST_to_GET', true);	//Prevent infinite internal redirect
```

Comment:
That's pretty weird in my opinion, but fine for now. Maybe I should take time to improve the whole way to return responses to make it easier and more clear. But what I have in mind would take a lot of time and ask a lot of changes, and so is error-prone.